Extra/Shaders/NewVegasReloaded/Effects/Includes/aioNAL_Utilities.hlsl

```diff
@@ -0,0 +1,84 @@
+/*
+float3 RGBtoHSV(float3 c)
+{
+	float4 K = float4(0.0, -1.0 / 3.0, 2.0 / 3.0, -1.0);
+	float4 p = lerp(float4(c.bg, K.wz), float4(c.gb, K.xy), step(c.b, c.g));
+	float4 q = lerp(float4(p.xyw, c.r), float4(c.r, p.yzx), step(p.x, c.r));
+
+	float d = q.x - min(q.w, q.y);
+	float e = 1.0e-10;
+	return float3(abs(q.z + (q.w - q.y) / (6.0 * d + e)), d / (q.x + e), q.x);
+}
+*/
+
+/*
+float3 HSVtoRGB(float3 c)
+{
+	float4 K = float4(1.0, 2.0 / 3.0, 1.0 / 3.0, 3.0);
+	float3 p = abs(frac(c.xxx + K.xyz) * 6.0 - K.www);
+	return c.z * lerp(K.xxx, saturate(p - K.xxx), c.y);
+}
+*/
+
+/*
+float randomNoise(in float3 uvw)
+{
+	float3 noise = (frac(sin(dot(uvw ,float3(12.9898,78.233, 42.2442)*2.0)) * 43758.5453));
+	return abs(noise.x + noise.y + noise.z) * 0.3333;
+}
+
+float linStep(float minVal, float maxVal, float t)
+{
+	return saturate((t - minVal) / (maxVal - minVal));
+}
+*/
+
+/////////////////////
+////Begin prod80 code
+/////////////////////			
+		
+	float grayValue(float3 gv)
+{
+   return dot( gv, float3(0.2125, 0.7154, 0.0721) );
+}
+
+float smootherstep(float edge0, float edge1, float x)
+{
+   x = clamp((x - edge0)/(edge1 - edge0), 0.0, 1.0);
+   return x*x*x*(x*(x*6 - 15) + 10);
+}
+
+float Hue(float3 color)
+{
+   float hue = 0.0f;
+   float fmin = min(min(color.r, color.g), color.b);
+   float fmax = max(max(color.r, color.g), color.b);
+   float delta = fmax - fmin;
+   
+   if (delta == 0.0)
+      hue = 0.0;
+   else
+   {         
+      float deltaR = (((fmax - color.r) / 6.0) + (delta / 2.0)) / delta;
+      float deltaG = (((fmax - color.g) / 6.0) + (delta / 2.0)) / delta;
+      float deltaB = (((fmax - color.b) / 6.0) + (delta / 2.0)) / delta;
+
+      if (color.r == fmax )
```

Comment:
maybe here we could use a couple of ternaries or lerps? maybe even a float3 to do all 3 calculations at once, then select the channel with a ternary